src/vxlan.cpp

```diff
@@ -0,0 +1,45 @@
+#ifdef _WIN32
+    #include <winsock2.h>
+#else // _WIN32
+    #include <arpa/inet.h>
+#endif // _WIN32
+#include <tins/internals.h>
+#include <tins/memory_helpers.h>
+#include <tins/vxlan.h>
+
+using Tins::Memory::InputMemoryStream;
+using Tins::Memory::OutputMemoryStream;
+
+namespace Tins {
+
+VXLAN::VXLAN(const uint32_t vni) {
+    header_.flags = 0x08000000;
+    header_.vni = vni << 8;
+}
+
+VXLAN::VXLAN(const uint8_t* buffer, uint32_t total_sz) {
+    InputMemoryStream stream(buffer, total_sz);
+    stream.read(header_);
+    header_.flags = ntohl(header_.flags);

```

Comment:
Typically the pattern all other PDUs follow is:
* The constructor from a buffer just copies things as-is. This means it will preserve the protocol's endianness. 
* `write_serialization` copies it as-is.
* `get_XYZ` and `set_XYZ` handle the conversion by using `Endian::host_to_be` and `Endian::be_to_host` (those handle the type as well so you don't have to call one specific to e.g. 32 bit ints).

I'm not saying the way you're doing it is wrong, but I'd prefer if all PDUs did it consistently.

Reply:
Sure thing. I have modified and moved things around to follow the standard PDU implementation. Do let me know if any further changes are required.